packages/app/cypress/e2e/integration/settings.spec.ts

```diff
@@ -1,99 +1,94 @@
-    cy.wait('@SetPreferred').its('request.body.variables.value').should('include', '/usr/local/bin/vim')
+      // navigate away and come back
+      // preferred editor selected from dropdown should have been persisted
+      cy.visitApp()
+      cy.get('[href="#/settings"]').click()
+      cy.wait(100)
+      cy.get('[data-cy="Device Settings"]').click()
 
-    cy.contains('Choose your editor...').click()
-    cy.contains('Well known editor').click()
-    cy.wait('@SetPreferred').its('request.body.variables.value').should('include', '/usr/bin/well-known')
+      cy.get('[data-cy="use-well-known-editor"]').should('be.checked')
+      cy.get('[data-cy="use-custom-editor"]').should('not.be.checked')
+    })
 
-    // navigate away and come back
-    // preferred editor selected from dropdown should have been persisted
-    cy.visitApp()
-    cy.get('[href="#/settings"]').click()
-    cy.wait(100)
-    cy.get('[data-cy="Device Settings"]').click()
+    it('allows custom editor', () => {
+      cy.intercept('POST', 'mutation-ExternalEditorSettings_SetPreferredEditorBinary').as('SetPreferred')
 
-    cy.get('[data-cy="use-well-known-editor"]').should('be.checked')
-    cy.get('[data-cy="use-custom-editor"]').should('not.be.checked')
+      // doing invoke instead of `type` since `type` enters keys on-by-one, triggering a mutation
+      // for each keystroke, making it hard to intercept **only** the final request, which I want to
+      // assert contains `/usr/local/bin/vim'
+      cy.findByPlaceholderText('Custom path...').clear().invoke('val', '/usr/local/bin/vim').trigger('input').trigger('change')
+      cy.wait('@SetPreferred').its('request.body.variables.value').should('include', '/usr/local/bin/vim')
 
-    // change to custom editor using input
-    cy.findByPlaceholderText('Custom path...').clear().invoke('val', '/usr/local/bin/vim').trigger('input').trigger('change')
+      // navigate away and come back
+      // preferred editor entered from input should have been persisted
+      cy.get('[href="#/settings"]').click()
+      cy.wait(100)
+      cy.get('[data-cy="Device Settings"]').click()
 
-    // navigate away and come back
-    // preferred editor entered from input should have been persisted
-    cy.get('[href="#/settings"]').click()
-    cy.wait(100)
-    cy.get('[data-cy="Device Settings"]').click()
+      cy.get('[data-cy="use-well-known-editor"]').should('not.be.checked')
+      cy.get('[data-cy="use-custom-editor"]').should('be.checked')
+      cy.get('[data-cy="custom-editor"]').should('have.value', '/usr/local/bin/vim')
+    })
+
+    it('lists file browser as available editor', () => {
+      cy.intercept('POST', 'mutation-ExternalEditorSettings_SetPreferredEditorBinary').as('SetPreferred')
 
-    cy.get('[data-cy="use-well-known-editor"]').should('not.be.checked')
-    cy.get('[data-cy="use-custom-editor"]').should('be.checked')
+      cy.contains('Choose your editor...').click()
```

Comment:
We should use the `defaultMessages` when we can i.e.

```
import defaultMessages from '@packages/frontend-shared/src/locales/en-US.json'
cy.contains(defaultMessages.settingsPage.editor.noEditorSelectedPlaceholder).click()
```

Reply:
This is something I always struggle with when writing tests - it feels strange to be importing values to test against.

On the one hand, there's less duplication of strings, and if the string changes, then we don't have to update the test. We try to avoid embedding strings in code for good reasons, and some of those reasons apply to embedding strings in tests too.

On the other hand, if the string changes... the behavior has changed and wouldn't we want the test to point that out? If I replaced `noEditorSelectedPlaceholder` with an empty string in the localization, the test would still pass despite not showing what we expected to the user.

Reply:
That's a fair point. Plus, it makes the test a bit harder to read since I don't know exactly what DOM element it's trying to find, I either have to wait for the test to run or dig through the json. I think this is a good point to bring up to the team since we aren't consistent either way

Reply:
I agree with @BlueWinds - I don't think it makes sense to assert against the same thing we test against. You could changing the string in en-US.json to some random junk and it would still pass.

Reply:
For anyone following along at home, we've started a discussion in Slack about this, currently waiting for Jess to return and give her opinion on the matter before deciding which direction to go.